ffi/jpeg.lua

```diff
@@ -63,24 +65,19 @@ function Jpeg.encodeToFile(filename, source_ptr, w, h, quality, color_type, subs
     color_type = color_type or turbojpeg.TJPF_RGB
     subsample = subsample or turbojpeg.TJSAMP_420
 
-
     local jpeg_size = ffi.new("unsigned long int [1]")
-
     local jpeg_image = ffi.new("unsigned char* [1]")
-    jpeg_image[0] = ffi.new("unsigned char*")
 
     local handle = turbojpeg.tjInitCompress()
     assert(handle, "no TurboJPEG API compressor handle")
 
     if turbojpeg.tjCompress2(handle, source_ptr, w, 0, h, color_type,
         jpeg_image, jpeg_size, subsample, quality, 0) == 0 then
 
-        local of, err = io.open(filename, "wb")
-        if err ~= nil then
-            return err
-        else
-            of:write(ffi.string(jpeg_image[0], jpeg_size[0]))
-            of:close()
+        local fhandle = C.open(ffi.cast("char *", filename), C.O_RDWR)
```

Comment:
We might be missing a few flags to make this saner, `O_CREAT`, `O_APPEND`, `O_TRUNC`.

Reply:
i.e.,

```diff
diff --git a/ffi-cdecl/posix_decl.c b/ffi-cdecl/posix_decl.c
index ea3ee18d..0c3cb307 100644
--- a/ffi-cdecl/posix_decl.c
+++ b/ffi-cdecl/posix_decl.c
@@ -27,6 +27,9 @@ cdecl_func(fork)
 cdecl_func(dup)
 cdecl_func(dup2)
 
+cdecl_const(O_APPEND)
+cdecl_const(O_CREAT)
+cdecl_const(O_TRUNC)
 cdecl_const(O_RDWR)
 cdecl_const(O_RDONLY)
 cdecl_const(O_NONBLOCK)
diff --git a/ffi/posix_h.lua b/ffi/posix_h.lua
index 999ce719..897c871a 100644
--- a/ffi/posix_h.lua
+++ b/ffi/posix_h.lua
@@ -16,6 +16,9 @@ int pipe(int *) __attribute__((nothrow, leaf));
 int fork(void) __attribute__((nothrow));
 int dup(int) __attribute__((nothrow, leaf));
 int dup2(int, int) __attribute__((nothrow, leaf));
+static const int O_APPEND = 1024;
+static const int O_CREAT = 64;
+static const int O_TRUNC = 512;
 static const int O_RDWR = 2;
 static const int O_RDONLY = 0;
 static const int O_NONBLOCK = 2048;
```

And use `O_RDWR | O_CREAT | O_TRUNC` here.

Reply:
(Updated diff above).

Reply:
Don't we want to use "+" instead of "|" in this case? The code would be more readable.

Reply:
The idiomatic C syntax is an OR, which, granted, mostly looks like an ADD in this context, but I'd rather not deviate too much from that ;).